src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -81,19 +82,27 @@ private SwitchBootstraps() {}
     private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
     private static final boolean previewEnabled = PreviewFeatures.isEnabled();
 
+    private static final ClassDesc CD_BiPredicate = referenceClassDesc(BiPredicate.class);
+    private static final ClassDesc CD_Objects = ReferenceClassDescImpl.ofValidated("Ljava/util/Objects;");
 
-    private static final MethodType TYPES_SWITCH_TYPE = MethodType.methodType(int.class,
+    private static final MethodTypeDesc CHECK_INDEX_DESCRIPTOR =
+            MethodTypeDescImpl.ofValidated(ConstantDescs.CD_int, ConstantDescs.CD_int, ConstantDescs.CD_int);
+    private static final MethodTypeDesc MTD_TYPE_SWITCH = MethodTypeDesc.of(ConstantDescs.CD_int,
+            ConstantDescs.CD_Object,
+            ConstantDescs.CD_int);
+    private static final MethodTypeDesc MTD_TYPE_SWITCH_EXTRA = MethodTypeDesc.of(ConstantDescs.CD_int,
+            ConstantDescs.CD_Object,
+            ConstantDescs.CD_int,
+            CD_BiPredicate,
+            ConstantDescs.CD_List);
```

Comment:
These can use `MethodTypeDescImpl​::ofValidated(…)`:
```suggestion
    private static final MethodTypeDesc MTD_TYPE_SWITCH = MethodTypeDescImpl.ofValidated(ConstantDescs.CD_int,
            ConstantDescs.CD_Object,
            ConstantDescs.CD_int);
    private static final MethodTypeDesc MTD_TYPE_SWITCH_EXTRA = MethodTypeDescImpl.ofValidated(ConstantDescs.CD_int,
            ConstantDescs.CD_Object,
            ConstantDescs.CD_int,
            CD_BiPredicate,
            ConstantDescs.CD_List);
```